contracts/authorization/src/contract.rs

```diff
@@ -953,6 +1085,82 @@ fn process_polytone_callback(
         .add_attribute("action", "process_polytone_callback"))
 }
 
+fn process_hyperlane_callback(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    handle_msg: HandleMsg,
+) -> Result<Response, ContractError> {
+    // We need to check that the callback comes from a registered processor address in the external domains and that the domain ID matches
+    // and obtain the encoder to decode the callback
+    let encoder = EXTERNAL_DOMAINS
+        .range(deps.storage, None, None, Order::Ascending)
+        .filter_map(Result::ok)
+        .find_map(|(_, external_domain)| {
```

Comment:
Do we really need the range here? seems like we should also keep record where the address is the key? 
Write once seems to be much better solution then loop on each callback.